servers/api_servers/routers/secrets.py

```diff
@@ -0,0 +1,91 @@
+"""
+This module provides API endpoints for managing secrets.
+
+Endpoints:
+    - GET /secrets/ : Retrieve all secrets as a list of key-value pairs.
+    - GET /secrets/{secret_key}/ : Retrieve a specific secret by its key.
+    - POST /secrets/{secret_key}/ : Create or update a secret with a given key and value.
+"""
+
+from fastapi import APIRouter, HTTPException
+from pydantic import BaseModel
+
+from servers.admin_ui.common import get_secret_provider
+
+router = APIRouter()
+
+
+class Secret(BaseModel):
```

Comment:
Will this be extensible for a future where environment variables are gathered by multiple secrets? Instead of just one that contains all of them. I think it's an important feature that we'll need to support soon